tracer/src/Datadog.Trace/Iast/IastModule.cs

```diff
@@ -631,6 +639,11 @@ private static IastModuleResponse GetScope(string evidenceValue, IntegrationId i
         {
             if (isRequest)
             {
+                if (traceContext?.RootSpan is { } rootSpan)
+                {
+                    rootSpan.Context.TraceContext?.SetSamplingPriority(SamplingPriorityValues.UserKeep, SamplingMechanism.Asm);
+                }
```

Comment:
Same comment as above.
```suggestion
                traceContext?.SetSamplingPriority(SamplingPriorityValues.UserKeep, SamplingMechanism.Asm);
```